adafruit_led_animation/animation/comet.py

```diff
@@ -57,7 +57,12 @@ class Comet(Animation):
     :param color: Animation color in ``(r, g, b)`` tuple, or ``0x000000`` hex format.
     :param int tail_length: The length of the comet. Defaults to 10. Cannot exceed the number of
                             pixels present in the pixel object, e.g. if the strip is 30 pixels
-                            long, the ``tail_length`` cannot exceed 30 pixels.
+                            long, the ``tail_length`` cannot exceed 30 pixels, or, in the case of a
+                            pixel mapped grid, if the width of the grid is 13 pixels, a horizontal
+                            comet tail cannot exceed 13 pixels. To find the lenth of the
+                            ``pixel_object``, include a ``print(len(PIXEL_OBJECT_NAME))`` in your
+                            code anywhere after the object creation, where ``PIXEL_OBJECT_NAME`` is
+                            the name assigned to the object, such as ``pixels``.
```

Comment:
How about:
- if `tail_length is `None` just set it to len(pixel_object)`
- otherwise set it to `tail_length = min(tail_length, len(pixel_object))`

e.g. `tail_length=None` in the `def`, then
```python
    if tail_length is None:
        tail_length = len(pixel_object)
    else:
        tail_length = min(tail_length, len(pixel_object))
```

Then there are no exceptions and the default is sensible.

Reply:
@dhalbert I like it. However, if I set the tail length to greater than the length of the pixel_object, it fails as follows:
```Auto-reload is on. Simply save files over USB to run them or enter REPL to disable.
code.py output:
Traceback (most recent call last):
  File "code.py", line 30, in <module>
  File "/lib/adafruit_led_animation/sequence.py", line 212, in animate
  File "/lib/adafruit_led_animation/animation/__init__.py", line 98, in animate
  File "/lib/adafruit_led_animation/animation/comet.py", line 143, in draw
  File "/lib/adafruit_led_animation/animation/comet.py", line 129, in _comet_generator
ValueError: Unmatched number of items on RHS (expected 32, got 33).
```
If I change it to:
```tail_length = min(tail_length, (len(pixel_object) - 1))```
it works.
Is this acceptable?

Reply:
I think this is because of:
https://github.com/adafruit/Adafruit_CircuitPython_LED_Animation/blob/a90235b31a43a851c72fe142255c289fcedb251e/adafruit_led_animation/animation/comet.py#L85. If you change that to (removing the `+ 1`
```python
         self._tail_length = tail_length
```
then maybe it will all work. without having to adjust the incoming `tail_length`.

Reply:
Tail length needs to be at least 2, so I added to your code. Took out the `+ 1` and it works without adjustment, as you said.

Reply:
Sounds good, I assume it looks the same!